src/common/tse.c

```diff
@@ -803,9 +797,6 @@ tse_task_complete(tse_task_t *task, int ret)
 	/** update task in scheduler lists. */
 	if (!dsp->dsp_cancelling && done)
 		tse_sched_process_complete(dsp);
-	/** If another thread canceled, make sure we drop the ref count */
-	else if (bumped)
```

Comment:
would this work if you actually just remove the else here?
the idea was that if someone else cancelled in the middle of completion, we would need to keep it around to finish here.

Reply:
> would this work if you actually just remove the else here?

I considered just removing "else", it should work. But seems to me that need not addref. 

> the idea was that if someone else cancelled in the middle of completion, we would need to keep it around to finish here.

you mean someone calls tse_sched_complete() with cancel flag as true then possibly free the dsp?
Here the task at least still hold one refcount for the dsp, seems need not take extra refcount for dsp?